src/main/java/ui/MenuControl.java

```diff
@@ -89,47 +86,69 @@ private Menu createFileMenu() {
     }
 
     /**
-     * Logs the user action and creates a new board
+     * Checks whether the current open board has unsaved changes
+     * @return {@code true} if current open board is dirty, {@code false} otherwise.
      */
-    private void onBoardNew() {
-        logger.info("Menu: Boards > New");
-        newBoard();
+    public final boolean isCurrentOpenBoardDirty() {
+        if (!prefs.getLastOpenBoardPanelInfos().isPresent()) {
+            return true;
+        }
+        return !panels.getCurrentPanelInfos().equals(prefs.getLastOpenBoardPanelInfos().get());
     }
 
     /**
-     * Creates an empty board. User will be prompted to
-     * confirm the action if there are unclosed panels
+     * Prompts a {@code ConfirmChangesDialog} to ask the user whether or not to save the unsaved changes
+     * in the current board.
      */
-    public final void newBoard() {
-        if (!isNewBoardCreationDialogConfirmed()) {
-            logger.info("New board creation cancelled");
+    public final void promptToSaveCurrentBoard() {
+        ConfirmChangesDialog dlg = new ConfirmChangesDialog(mainStage);
+        Optional<ButtonType> response = dlg.showAndWait();
+        if (response.get() != ButtonType.YES) {
+            logger.info("User abandoned unsaved changes.");
             return;
         }
 
-        panels.closeAllPanels();
-        saveBoardAs();
+        saveBoard();
+        logger.info("Changes to the current board saved.");
     }
 
     /**
-     * Returns a boolean indicating whether we can proceed with
-     * the new board creation. If there are no panels open, it
-     * returns true; else a dialog is shown for the user to confirm.
+     * Prompts a {@code BoardNameDialog} to ask the user for a new board name
+     * @return the user's response
      */
-    private boolean isNewBoardCreationDialogConfirmed() {
-        List<PanelInfo> panelList = panels.getCurrentPanelInfos();
-        if (panelList.isEmpty()) {
-            return true;
+    public final Optional<String> promptForNewBoardName() {
+        BoardNameDialog dlg = new BoardNameDialog(prefs, mainStage);
+        Optional<String> response = dlg.showAndWait();
+        ui.showMainStage();
+        panels.selectFirstPanel();
+        return response;
+    }
+
+    /**
+     * Logs the user action and creates a new board
+     */
+    private void onBoardNew() {
```

Comment:
This is the method that will be the called when user clicked `Boards > New`. It is named similarly as other user click actions. 

I abstract all the actual `creating`, `deleting` and `saving` actions out of the UI event handlers in #1462 to make them available as APIs. 

The only arguable thing here is whether I should put the prompt in the UI event handler (i.e. `onBoardNew`, it is not really an event handler but it acts like an event handler), or in the actual action. I put it in `newBoard` instead of `onBoardNew` because I think some other classes that will be calling `newBoard` might not want that prompt.